### DISCUSSION TITLE
Is it possible to recompile peachpie DLL to PHP?

### DISCUSSION THREAD

**Discussion by noobsoftware** [peachpiecompiler/peachpie#1045] [Q&A] [answered]
I have a nuget with DLLs compiled from PHP using peachpie, I'm wondering if i can recover the original PHP files or recompile to PHP?

**Comment by jakubmisek** [comment, marked as answer]
DLL contains the code compiled into MSIL. You can try to decompile (https://docs.peachpie.io/scenarios/intermediate/decompilation/) into C# to see at least what classes and functions are there. Although you won't obtain the original source code.

**Reply by noobsoftware** [reply]
I'm able to decompile to C# but i was hoping for there to be PHP as resource files or something like that. The problem being the DLLs and nuget are using .NetStandard2.1 but i need .NETStandard 2.0, but i can hopefully make do with the C# code.

**Reply by jakubmisek** [reply]
You can target `netstandard2.0` as well in your [project file's TargetFramework](https://docs.peachpie.io/php/msbuild/). 

We're just putting `netstandard2.1` as the default in our samples and new projects templates.

**Reply by noobsoftware** [reply]
I found that version 0.9.990 does not have mono.posix.netstandard which means it works for me on Xamarin.Mac, I'm wondering are there still plans to remove Mono.Posix, because when i try to install the latest version in my mac xamarin project i get this:

For adding package 'Mono.Posix.NETStandard.1.0.0' to project 'numeval.macOS' that targets 'xamarinmac20'.
Install failed. Rolling back...
Package 'Mono.Posix.NETStandard 1.0.0' does not exist in project 'numeval.macOS'

I did manage to use a workaround for installing on a Xamarin WPF project, but i am still unable to get this working on Xamarin.mac

**Reply by jakubmisek** [reply]
`0.9` is extremely old ..

According to https://www.nuget.org/packages/Mono.Posix.NETStandard/ - Frameworks - Xamarin.* should be supported. 

I remember I had to install "Mono.Posix" as well in order to make this work on Xamarin.

**Reply by noobsoftware** [reply]
Strange: Could not install package 'Mono.Posix.NETStandard 1.0.0'. You are trying to install this package into a project that targets 'Xamarin.Mac,Version=v2.0', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.

I will i create an issue for this i suppose.

**Reply by jakubmisek** [reply]
Try installing "Mono.Posix"
